tests.py

```diff
@@ -0,0 +1,303 @@
+from os.path import join
```

Comment:
How do you run the tests?
Do you have a docker-compose that brings up a the minio server against the tests code?

Reply:
currently locally with minio docker, im going to add the ci this as well in the close future